cookbooks/bcpc-hadoop/templates/default/hdp_mapred-site.xml.erb

```diff
@@ -64,4 +64,30 @@
     <name>yarn.app.mapreduce.am.staging-dir</name>
     <value>/user</value>
   </property>
+
+  <property> 
+      <name>mapreduce.map.memory.mb</name> 
+      <value><%= node['bcpc']['hadoop']['yarn']['scheduler']['minimum-allocation-mb'] %></value> 
+  </property> 
+
+  <property> 
+      <name>mapreduce.map.java.opts</name> 
+      <value><%="-Xmx #{0.8 * node['bcpc']['hadoop']['yarn']['scheduler']['minimum-allocation-mb']}" %></value> 
```

Comment:
Ray, the 80% comes from the [Hortonworks manual tuning guide](http://docs.hortonworks.com/HDPDocuments/HDP2/HDP-2.0.9.1/bk_installing_manually_book/content/rpm-chap1-11.html).